hazelcast/src/test/java/com/hazelcast/quorum/QuorumListenerTest.java

```diff
@@ -229,4 +257,16 @@ public void onChange(QuorumEvent quorumEvent) {
         }
         assertOpenEventually(belowLatch);
     }
+
+    static class QuorumListener implements com.hazelcast.quorum.QuorumListener {
+
+        static AtomicInteger failCount = new AtomicInteger();
+        @Override
+        public void onChange(QuorumEvent quorumEvent) {
+            assertEquals(3, quorumEvent.getThreshold());
```

Comment:
I don't know if this is going to work. Because it will throw an Assertion error on the thread that calls the listener; not the thread that calls the test.

The if(.. isPresent) is the right approach to deal with the asynchronous behavior.